agent/agent.c

```diff
@@ -20,23 +20,199 @@
 
 #include "agent.h"
 
+#include <hwangsae/dbus/manager-generated.h>
+#include <hwangsae/dbus/edge-interface-generated.h>
+#include <chamge/chamge.h>
+
+#define DEFAULT_HUB_UID        "abc-987-123"
+#define DEFAULT_BACKEND         CHAMGE_BACKEND_AMQP
+
 struct _HwangsaeAgent
 {
   GApplication parent;
+  Hwangsae1DBusManager *manager;
+  Hwangsae1DBusEdgeInterface *edge_interface;
+  ChamgeHub *chamge_hub;
 };
 
 /* *INDENT-OFF* */
 G_DEFINE_TYPE (HwangsaeAgent, hwangsae_agent, G_TYPE_APPLICATION)
 /* *INDENT-ON* */
 
+static gboolean
+hwangsae_agent_dbus_register (GApplication * app,
+    GDBusConnection * connection, const gchar * object_path, GError ** error)
+{
+  gboolean ret;
+  HwangsaeAgent *self = HWANGSAE_AGENT (app);
+
+  g_debug ("hwangsae_agent_dbus_register");
+
+  g_application_hold (app);
+
+  /* chain up */
+  ret =
+      G_APPLICATION_CLASS (hwangsae_agent_parent_class)->dbus_register
+      (app, connection, object_path, error);
+
+  if (ret &&
+      !g_dbus_interface_skeleton_export (G_DBUS_INTERFACE_SKELETON
+          (self->manager), connection,
+          "/org/hwangsaeul/Hwangsaeul1/Manager", error)) {
```

Comment:
Perhaps, it is `/org/hwangsaeul/Hwangsae1/Manager`. 
What do you think the naming of interface? This is a very Korean name. Is it an inconvenience?

Reply:
I don't see any problem with the name for being Korean, so I think that we should go with your suggestion.